getting_started/step_by_step/7_room_two.rst

```diff
@@ -0,0 +1,211 @@
+* From ``backgrounds.zip``, copy the ``park_bg.png`` file to the **rooms**
+  directory and set it as the ``texture`` parameter in the ``ESCBackground``
+  node.
+* Create a walkable area for the character. Add an ``ESCTerrain``
+  node to the ``ESCRoom`` node. Add a
+  ``NavigationPolygonInstance`` node as a child of the new ``ESCTerrain`` node.
+* Create a NavigationPolygonInstance resource for the Navpoly parameter.
+
+.. image:: img/create_room_terrain_navpoly2.png
+   :alt: Adding a NavPoly
+
+* Use the editor window to draw a polygon. This polygon is the area in which
+  the character can move around freely.
+
+.. image:: img/create_room_terrain_createdpoly2.png
+   :alt: View of the completed polygon on the background
+
+Creating a light map
+--------------------
+
+For this room we're going to use a light map. As mentioned previously, light
+maps are used to shade different areas of the room differently. In this room,
+we have a lamp, so we want the player to be brightly lit when under it, and
+less brightly lit when away from the light source.
+
+The example light map provided is a very simple one with just 3 colours to make
+it easier to demonstrate how the light map changes how the game character
+looks, but you could replace it with one that uses a gradient to fade nicely
+from white to mid-grey. Coloured ones can also be used to simulate things like
+concert or traffic lights.
+
+Copy the park_lightmap.png asset (link on the main step-by-step page) to the
+room02 folder.
+
+In the Godot GUI, select the ``ESCTerrain`` node, and set its Lightmap to the
+park_lightmap.png texture.
+
+Create a doorway back to the pub room
+-------------------------------------
+To link room 2 (the park) to room 1 (the pub), you will need an ESCexit. When
+the player clicks this, it tells Escoria that you want to change rooms.
+
+Under the ESCroom node, create an ESCexit node.
+
+.. image:: img/create_room_esc_exit.png
+   :alt: The ESC exit node.
+
+Set its Global ID to "exit_to_pub"
+
+Create a CollisionPolygon2D as a child of the ESCexit. Use the editor window to
+draw a rectangle covering the left-hand-edge of the park background. This
+polygon is the area which will act like a "door" for the player to click on to
+change rooms. (If the room had a doorway, you'd draw the polygon around the
+door).
+
+.. image:: img/create_room_park_exit.png
+   :alt: The ESCexit polygon.
+
+When the player clicks on the this area, Escoria needs to know what to do, for
+this we will use a script. Firstly, Escoria will need to know where to make
+the character walk to when the user clicks on this polygon to leave the room.
+Create an ESClocation as a child of the ESCexit. Give this location the
+``Global ID`` of "park_exit".
+
+.. image:: img/create_room_exit_location.png
+   :alt: The doorways exit point.
+
+Now as in the previous examples, we need to link the ESCexit to a script.
+
+Create a doorway script
```

Comment:
ESCExit is streamlined so you actually don't need that. Check out its doc: https://docs.escoria-framework.org/en/devel/api/ESCExit.html

You just need to set the target scene and can additionally set a "switch sound".

Reply:
So it does the change_scene command for you? If that's the case then the documentation page for ESCExit should explicitly say that (I've got an outstanding documentation issue raised where I mention that the ESCExit documentation needs more detail).
So if you can clarify the differences, I'll update both this documentation and the ESCExit documentation.
Am I right in saying that if the only thing you want to happen when you use a "door" is a "change_scene" command (and optional sound) you use "ESCexit", but if you wanted something else to happen (e.g. animate an NPC, say something as you leave etc) you'd use an ESCItem with the exit button selected and attach a script?

Reply:
Yes. I thought it would be understandable by the name and parameters, but you're right. 🙂 

And yes, that's a good description when to use or not to use ESCExit.

Reply:
Rewrote this section to use ESCExit rather than ESCItem so this should now  be correct.